docs/spec_draft.md

```diff
@@ -348,6 +349,93 @@ logical operation.
 
 [Back to Ops](#index-of-ops)
 
+## stablehlo.batch_norm_training
+
+### Semantics
+
+Computes mean and variance across all dimensions except for the `feature_index`
+dimension and normalizes the `operand` tensor producing `output`, `batch_mean`
+and `batch_var` tensors. More formally, this operation can be expressed as a
+decomposition to existing StableHLO operations using Python-like syntax as
+follows:
+
+```python
+def compute_mean(operand, feature_index):
+  (sum,) = reduce(
+      inputs=[operand],
+      init_values=[0],
+      dimensions=[i for i in range(rank(operand)) if i != feature_index],
+      body=lambda x, y: add(x, y))
+  divisor = constant(num_elements(operand) / dim(operand, feature_index))
+  divisor_bcast = broadcast_in_dim(divisor, [], shape(sum))
+  return divide(sum, divisor_bcast)
+
+def compute_variance(operand, mean, feature_index):
+  mean_bcast = broadcast_in_dim(mean, [feature_index], shape(operand))
+  centered_operand = subtract(operand, mean_bcast)
```

Comment:
would it be better if we wrote `stablehlo.broadcast_in_dim` and `stablehlo.subtract` etc
this was a suggestion from @sdasgup3

Reply:
I think that in the context of a pseudocode in the StableHLO spec it's fine to not qualify these functions with `stablehlo.`. Unless there are naming conflicts with meta-functions like `shape` or `num_elements`, which we don't have at the moment.